### PR TITLE
Add `set_local` support for static quant with pt2e

### PR DESCRIPTION
**Pull request by yiliu30**
## Type of Change

feature
API change: enable `set_local`

## Description

Require torch >= 2.3.2
 
```python
quant_config = StaticQuantConfig()
quant_config.set_local("fc1", StaticQuantConfig(w_dtype="fp32", act_dtype="fp32"))
q_model = quantize(model=model, quant_config=quant_config, run_fn=calib_fn)
```

## Expected Behavior & Potential Risk

the expected behavior that triggered by this PR 

## How has this PR been tested?

how to reproduce the test (including hardware information)

## Dependency Change?

any library dependency introduced or removed

### CODE REVIEW

**Comment by yiliu30** [comment]
https://dev.azure.com/lpot-inc/b7121868-d73a-4794-90c1-23135f974d09/_apis/build/builds/31463/logs/20

**Review by Kaihui-intel — APPROVED** [review]

File: neural_compressor/torch/algorithms/pt2e_quant/utility.py [hunk 1]
```diff
@@ -75,5 +80,22 @@ def create_xiq_quantizer_from_pt2e_config(config, is_dynamic=False) -> X86Induct
     # set global
     global_config = _map_inc_config_to_torch_quant_config(config, is_dynamic)
     quantizer.set_global(global_config)
-    # Skip the local config for now (need torch 2.4)
+    # need torch >= 2.3.2
```
Comment (yiliu30) [inline neural_compressor/torch/algorithms/pt2e_quant/utility.py:83]: As it requires torch > 2.3.2, there are several `# pragma: no cover`s.

File: neural_compressor/torch/utils/environ.py [hunk 2]
```diff
@@ -91,6 +91,9 @@ def get_torch_version():
     return version
 
 
+GT_TORCH_VERSION_2_3_2 = get_torch_version() > Version("2.3.2")
```
Comment (xin3he) [inline neural_compressor/torch/utils/environ.py:94]: How about following transformers design? `check_min_torch_version("2.3.2")` or `require_version("torch>=2.3.2", "log")`
```
# Will error if the minimal version of Transformers is not installed. Remove at your own risks.
check_min_version("4.42.0.dev0")

require_version("datasets>=2.14.0", "To fix: pip install -r examples/pytorch/language-modeling/requirements.txt")
```

**Review by xin3he — APPROVED** [review]

File: neural_compressor/torch/utils/environ.py (hunk 2 above)
Comment (yiliu30) [reply neural_compressor/torch/utils/environ.py:94]: Actually, this is a temporary flag check. We can remove it after the Torch 2.3.2 or 2.4 release, which is coming soon.

**Pull request merged**